src/app/reporting/Engine.cpp

```diff
@@ -124,7 +124,11 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportDataMessage::Bui
             {
                 if (clusterInfo->IsAttributePathSupersetOf(*path))
                 {
-                    err = RetrieveClusterData(apReadHandler->GetFabricIndex(), attributeDataList, *path);
+                    // SetDirty injects path into GlobalDirtySet path that don't have the particular nodeId,
+                    // need to inject nodeId from subscribed path here.
+                    ClusterInfo dirtyPath = *path;
+                    dirtyPath.mNodeId = clusterInfo->mNodeId;
```

Comment:
Apologies.  I'm confused.  Is the ClusterInfo mNodeId the local node ID, or the node ID of the peer?  If it's the local node ID, doesn't fabric index also give us this?

Reply:
the node id is current server application local nodeId. may I know how fabric index offer node Id info? When constructing attribute path in report data, we need to fill actual nodeId in tlv.

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Encoding-Specification.adoc#encoding-AttributePathIB-Node

Reply:
currently we are putting nodeId in attribute path.

Reply:
Note, though, that except for proxies the Node id can (and should!) be omitted.

Reply:
yep, agree, still think about it regarding how to check proxy and skip node id.

Reply:
https://github.com/project-chip/connectedhomeip/issues/11513